File: 3974-xor-after-range-multiplication-queries-i/xor-after-range-multiplication-queries-i.py
class Solution:
    def xorAfterQueries(self, nums: List[int], queries: List[List[int]]) -> int:
        M = int(1e9 + 7)

        for query in queries:
            l, r, k, v = query
            while l <= r:
                nums[l] = (nums[l] * v) % M
                l += k

        result = 0
        for num in nums:
            result ^= num

        return result

File: 3974-xor-after-range-multiplication-queries-i/xor-after-range-multiplication-queries-i.js
/**
 * @param {number[]} nums
 * @param {number[][]} queries
 * @return {number}
 */
var xorAfterQueries = function (nums, queries) {
    const M = 1e9 + 7
    for (const query of queries) {
        let [l, r, k, v] = query
        while (l <= r) {
            nums[l] = (nums[l] * v) % M
            l += k
        }
    }
    let result = 0
    for (const num of nums) {
        result ^= num
    }
    return result
};